### ISSUE TITLE
OpenDOS 7.02 causes crash when *not* ran from a harddisk image

### ISSUE THREAD

**Issue by jschwartzenberg** [stsp/dosemu2#250]
I tested with latest git. A harddisk image containing OpenDOS can be found here:
ftp://ftp.nvg.org/pub/mirrors/metalab.unc.edu/distributions/caldera/updates/OpenLinux/1.3/current/SRPMS/opendos-hdimage-7.02-4.src.rpm $DESTINATION/opendos-hdimage-7.02-4.src.rpm

Inside the RPM will be a file hdimage.od. This file can be booted without problems by putting it inside ~/.dosemu/drives (dosemu prints an error but it works fine).

Copying the files from out of that image inside the host filesystem causes a crash however. I suspect this is because in the first case the OpenDOS MBR is used. I have attached logs with both kvm & emulated settings.
[boot.log.1.txt](https://github.com/stsp/dosemu2/files/573138/boot.log.1.txt)
[boot.log.txt](https://github.com/stsp/dosemu2/files/573139/boot.log.txt)

**Comment by stsp** [comment]
Did this ever work before, or are you testing
the new feature of using the raw hdimages?

@andrewbird where was that table you posted
containing the status of different DOS support?
I guess we need to put that into wiki or docs,
otherwise I can't remember what do we support
today. :)

**Comment by andrewbird** [comment]
Yes it was booting from fatfs based directory, (fixed in #151) and still is according to my test just now. See below.
~~~
Test DR-DOS-7.02 FAT12 vfs directory ... ok
Test DR-DOS-7.02 FAT16 vfs directory ... ok
Test DR-DOS-7.02 FAT16B vfs directory ... ok
Test DR-DOS-7.02 MFS redirection ... ok
Test DR-DOS-7.02 SysType ... ok
Test DR-DOS-7.02 VBFLPY ... ok
~~~

The sha1sums for the DR-DOS 7.02 I'm testing are:
~~~
        cls.version = "Caldera DR-DOS 7.02"                                     
        cls.tarfile = "DR-DOS-7.02.tar"                                         
        cls.files = [                                                           
            ("ibmbio.com", "af05e014d894cf4fda37d629446525e7018b7b17"),         
            ("ibmdos.com", "39ed49c4b6c3641061087b0b799c42fbc41b129c"),         
            ("command.com", "5757dc7604e12fb3bbfe96e53a37028060bcedf3"),        
        ]                                                                       
~~~
But the ones in the rpm you have are
~~~
63bad7432ae39d33eb194e94b5a8bfe7f32da003  ibmdos.com
902f8cae012445844f0e32b0effc190c8f52b33b  ibmbio.com
799123491276c13f35757cb2f2a9beb4a23237c5  command.com
~~~
So previous reports of DR-DOS 7.02 working do not apply to the case you have :-( 
I know the versioning between real DR-DOS and Enhanced DR-DOS can be confusing (at least to me anyway), but I don't think that's the issue here as both flavours appear to be genuine according to the strings in ibmbio.com

Mine
~~~
NaMe SeRvIcE=DR-DOS BIOS kernel
CoPyRiGhT=Copyright (C) 1983-1998 Caldera, Inc. All rights reserved. 
VeRsIoN=7.02 (980123)
LaNgUaGe=English (1)
SeRiAlNuMbEr=XXXX-0000-654321
~~~

Yours
~~~
NaMe SeRvIcE=OpenDOS BIOS kernel
CoPyRiGhT=Copyright (C) 1983-1997 Caldera, Inc. All rights reserved. 
VeRsIoN=7.02 (971119)
LaNgUaGe=English (1)
SeRiAlNuMbEr=XXXX-0000-654321
~~~

[Wikipedia](https://en.wikipedia.org/wiki/DR-DOS) has this text "After beta releases in September and November 1997, the next official release came in December 1997, with the name changed to Caldera DR-OpenDOS 7.02, soon followed by a further release in March 1998, when the DR-DOS name returned as Caldera DR-DOS 7.02, now for the first time written with a hyphen. Version 7.02 (now reporting itself as BDOS 7.3) incorporated improved BIOS and BDOS issues"

@stsp The table is buried somewhere in the closed issues I think, I had a quick look but can't spot it anywhere. I really hope to be able to generate this automatically sometime. Unfortunately I do get distracted easily!

@jschwartzenberg I'm going to have a fiddle with rpm image sourced ones here and see if I can spot some problem with dosemu.
